drivers/vulkan/rendering_device_vulkan.h

```diff
@@ -1145,6 +1145,7 @@ class RenderingDeviceVulkan : public RenderingDevice {
 	virtual void draw_list_enable_scissor(DrawListID p_list, const Rect2 &p_rect);
 	virtual void draw_list_disable_scissor(DrawListID p_list);
 
+	virtual uint32_t get_current_subpass();
```

Comment:
I would avoid this, the API is designed to not get information back for simplicity of portability, keep it yourself locally.

Reply:
@reduz The problem is that there is code where I don't know what subpass we're running especially once we make different breaks so we can support reads in the transparent pass. 

For instance if we don't need reads we can optimise by doing it as we do now:
- Pass 1 subpass 1 = Opaque
- Pass 1 subpass 2 = Sky
- Pass 1 subpass 3 = Transparent
- Pass 1 subpass 4 = Tonemap and copy to 2D buffer

Then if we have effects enabled that require reads we change it to:
- Pass 1 subpass 1 = Opaque
- Pass 1 subpass 2 = Sky
- Pass 2 subpass 1 = Transparent
- Pass 2 subpass 2 = Tonemap and copy to 2D buffer

And in the least optimal case, if we have glow or DOF turned on, or if we're rendering at half resolution, we end up with:
- Pass 1 subpass 1 = Opaque
- Pass 1 subpass 2 = Sky
- Pass 2 subpass 1 = Transparent
- Pass 3 subpass 1 = Tonemap and copy to 2D buffer

So we can see that for our transparent and tonemap/copy pass we can't hardcode the subpass, it changes depending on configuration.

Reply:
Obviously the alternative is to change render device that it uses the current subpass automatically so we don't need to pass it here.

Reply:
Ok, I guess it makes sense, I would then still rename the function to draw_list_get_current_pass(); to keep naming homogeneous.

Reply:
Done